src/Microsoft.DotNet.Darc/src/Darc/Models/PopUps/UpdateSubscriptionPopUp.cs

```diff
@@ -0,0 +1,225 @@
+                IDeserializer serializer = new DeserializerBuilder().Build();
+                outputYamlData = serializer.Deserialize<SubscriptionData>(yamlString);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Failed to parse input yaml.  Please see help for correct format.");
+                return Constants.ErrorCode;
+            }
+
+            _yamlData.Batchable = ParseSetting(outputYamlData.Batchable, _yamlData.Batchable, false);
+            _yamlData.Enabled = ParseSetting(outputYamlData.Enabled, _yamlData.Enabled, false);
+            // Make sure Batchable and Enabled are valid bools
+            if (!bool.TryParse(outputYamlData.Batchable, out bool batchable) || !bool.TryParse(outputYamlData.Enabled, out bool enabled))
+            {
+                _logger.LogError("Either Batchable or Enabled is not a valid boolean values.");
+                return Constants.ErrorCode;
+            }
+
+            if (!bool.Parse(outputYamlData.Batchable) && outputYamlData.MergePolicies == null)
+            {
+                return Constants.ErrorCode;
+            }
+
+            // Validate the merge policies
+            if (outputYamlData.MergePolicies != null &&!ValidateMergePolicies(outputYamlData.MergePolicies))
+            {
+                return Constants.ErrorCode;
+            }
+
+            _yamlData.MergePolicies = outputYamlData.MergePolicies;
+
+            // Parse and check the input fields
+            _yamlData.Channel = ParseSetting(outputYamlData.Channel, _yamlData.Channel, false);
+            if (string.IsNullOrEmpty(_yamlData.Channel))
+            {
+                _logger.LogError("Channel must be non-empty");
+                return Constants.ErrorCode;
+            }
+
+            _yamlData.SourceRepository = ParseSetting(outputYamlData.SourceRepository, _yamlData.SourceRepository, false);
+            if (string.IsNullOrEmpty(_yamlData.SourceRepository))
+            {
+                _logger.LogError("Source repository URL must be non-empty");
+                return Constants.ErrorCode;
+            }
+
+            _yamlData.UpdateFrequency = ParseSetting(outputYamlData.UpdateFrequency, _yamlData.UpdateFrequency, false);
+            if (string.IsNullOrEmpty(_yamlData.UpdateFrequency) || 
+                !Constants.AvailableFrequencies.Contains(_yamlData.UpdateFrequency, StringComparer.OrdinalIgnoreCase))
+            {
+                _logger.LogError($"Frequency should be provided and should be one of the following: " +
+                    $"'{string.Join("', '",Constants.AvailableFrequencies)}'");
+                return Constants.ErrorCode;
+            }
+
+            return Constants.SuccessCode;
+        }
+
+        /// <summary>
+        /// Helper class for YAML encoding/decoding purposes.
+        /// This is used so that we can have friendly alias names for elements.
+        /// </summary>
+        private class SubscriptionData
```

Comment:
I'd add an Id element here.  We may be able to enable a case where we can mass-edit subscriptions.